src/server/project.ts

```diff
@@ -878,13 +875,34 @@ namespace ts.server {
                 return;
             }
 
+            // Search our peer node_modules, then any globally-specified probe paths
+            // ../../.. to walk from X/node_modules/typescript/lib/tsserver.js to X/node_modules/
+            const searchPaths = [combinePaths(host.getExecutingFilePath(), "../../.."), ...this.projectService.pluginProbeLocations];

```

Comment:
i would do all the node-process-specific work in server.ts. we publish this as a library s well, and users like WebStorm might have a different layout.